### ISSUE TITLE
Very quiet music is beneath the Bluetooth “floor”.

### ISSUE THREAD

**Issue by tfabris** [tfabris/BlueGigaEmpeg#72]
Songs which start out very quietly and fade in slowly from nothing, do not immediately make sound. 

Bluetooth encoder appears to have a “floor”, below which the chip doesn’t even bother to encode and send audio at all. 

This isn’t a noise or gain problem because the audio is a perfect digital bitstream, and also the problem exists regardless of whether you are using digital or analog audio. 

The floor is really low. Only the quietest of sounds drops below it. Example tracks would be the beginning of Dark Side of the Moon, or the start of The Camera Eye. In those cases the audio kicks in only a few seconds after the actual track start. So it’s not a particularly bad problem, and isn’t usually noticeable even on the poster child songs. 

To do:
- research to find out if this is normal for Bluetooth A2DP audio. 
- see if you can get the AAC codec working and if that solves it. 
- see if you can find other iWrap commands which maybe lower that floor. Gain commands will not affect it since we are not using the analog gain stages at all. The docs are clear on this point: "In WT32i, the gain setting only affects the internal ADC & DAC and is ignored for I2S and SPDIF interfaces."

**Comment by tfabris** [comment]
Looked for other iWrap commands which might affect this, could not find any.

Whether or not this is "normal" for the Bluetooth SBC codec (Sub Band Codec, the default), is not really relevant if there are no commands in iWrap which let me modify it.

To do: Take another stab at APTXLL codec, and if you can get it working, then see if it fixes this issue.

Don't bother with AAC codec since the docs say that only apple devices support it.

**Comment by tfabris** [comment]
Awaiting: Arrival of the CSR USB-SPI programmer so that I can configure the address and license of the BlueGiga chip on one of the unfried Betz boards.

**Comment by tfabris** [comment]
Received CSR USB-SPI programmer and it works. I now have one board which I have put a test license for APTXLL onto. Sadly, it seems as though APTXLL is not supported by my Honda car stereo so APTXLL is a non starter. However I still need to perform the controlled experiments to determine if the APTXLL solves the "noise floor" problem. But now I have to do it on other gear that supports APTXLL. Since APTXLL didn't work on the Honda, I haven't been able to do a controlled test yet. Will need to poke at this a bit more on the Etekcity module or perhaps some other module, before giving up on it.

**Comment by tfabris** [comment]
To Do:
- Do a solid test of the noise floor with APTX-LL codec.
- You can tell which codec is being used because you can see a message like this at connection time:
``` A2DP CODEC SBC JOINT_STEREO 44100 BITPOOL 2-53 ```

**Comment by tfabris** [comment]
Nnnnnnnnnnnnope.

I did a good test while connected with APTX Low Latency mode. There is still a significant "floor" to the bluetooth audio. The floor is exactly the same as in the SBC codec. "The Camera Eye" doesn't make any sound until about 3.5 seconds into the track in APTX LL mode. 

Though I was able to get APTX-LL working, the entire process of installing it on the chip was so onerous that even if the licenses from Silicon Labs were dirt cheap, it's not even worth the hassle of trying to obtain the licenses at all, it's not even worth the phone call.

It'd be a different story if we were making a gaming system where latency was critical. It's not critical here.

Another reason to leave the existing SBC codec latency is that the track titles take a moment to show up on the car stereo screen, and the codec latency is real nice because it slows down the audio to be closer to the moment the track titles show up most of the time, so that's kind of a plus.

Finally, APTX-LL is not supported on my own Honda car stereo, so this is useless to me overall.

Closing this issue and giving up on APTX-LL for now.
